File: apps/user-mobile/modules/chime-sdk/ios/ChimeVideoView.swift
import ExpoModulesCore
import AmazonChimeSDK
import UIKit

/// Native video view for rendering Chime video tiles
class ChimeVideoView: ExpoView {
    let videoRenderView: DefaultVideoRenderView

    required init(appContext: AppContext? = nil) {
        videoRenderView = DefaultVideoRenderView()
        super.init(appContext: appContext)

        videoRenderView.translatesAutoresizingMaskIntoConstraints = false
        addSubview(videoRenderView)

        NSLayoutConstraint.activate([
            videoRenderView.topAnchor.constraint(equalTo: topAnchor),
            videoRenderView.bottomAnchor.constraint(equalTo: bottomAnchor),
            videoRenderView.leadingAnchor.constraint(equalTo: leadingAnchor),
            videoRenderView.trailingAnchor.constraint(equalTo: trailingAnchor)
        ])

        // Default black background
        backgroundColor = .black
        videoRenderView.backgroundColor = .black
    }

    func setScalingType(_ scalingType: String) {
        switch scalingType {
        case "aspectFill":
            videoRenderView.contentMode = .scaleAspectFill
        default:
            videoRenderView.contentMode = .scaleAspectFit
        }
    }

    func setMirror(_ mirror: Bool) {
        if mirror {
            videoRenderView.transform = CGAffineTransform(scaleX: -1, y: 1)
        } else {
            videoRenderView.transform = .identity
        }
    }
}

File: apps/user-mobile/modules/chime-sdk/android/src/main/java/expo/modules/chimesdk/ChimeVideoView.kt
package expo.modules.chimesdk

import android.content.Context
import android.graphics.Color
import android.widget.FrameLayout
import expo.modules.kotlin.AppContext
import expo.modules.kotlin.views.ExpoView
import com.amazonaws.services.chime.sdk.meetings.audiovideo.video.DefaultVideoRenderView
import com.amazonaws.services.chime.sdk.meetings.audiovideo.video.VideoScalingType

/**
 * Native video view for rendering Chime video tiles
 */
class ChimeVideoView(context: Context, appContext: AppContext) : ExpoView(context, appContext) {
    val videoRenderView: DefaultVideoRenderView

    init {
        videoRenderView = DefaultVideoRenderView(context)
        videoRenderView.layoutParams = FrameLayout.LayoutParams(
            FrameLayout.LayoutParams.MATCH_PARENT,
            FrameLayout.LayoutParams.MATCH_PARENT
        )

        addView(videoRenderView)

        // Default black background
        setBackgroundColor(Color.BLACK)
    }

    fun setScalingType(scalingType: String) {
        when (scalingType) {
            "aspectFill" -> videoRenderView.scalingType = VideoScalingType.AspectFill
            else -> videoRenderView.scalingType = VideoScalingType.AspectFit
        }
    }

    fun setMirror(mirror: Boolean) {
        videoRenderView.mirror = mirror
    }
}
